dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.cs

```diff
@@ -1459,21 +1178,23 @@ private static void ValidateMaxTokens(int? maxTokens)
         }
     }
 
-    private static void ValidateAutoInvoke(bool autoInvoke, int resultsPerPrompt)
+    private static async Task<T> RunRequestAsync<T>(Func<Task<T>> request)
     {
-        if (autoInvoke && resultsPerPrompt != 1)
+        try
         {
-            // We can remove this restriction in the future if valuable. However, multiple results per prompt is rare,
-            // and limiting this significantly curtails the complexity of the implementation.
-            throw new ArgumentException($"Auto-invocation of tool calls may only be used with a {nameof(AzureOpenAIPromptExecutionSettings.ResultsPerPrompt)} of 1.");
+            return await request.Invoke().ConfigureAwait(false);
+        }
+        catch (RequestFailedException e)
+        {
+            throw e.ToHttpOperationException();
         }
     }
 
-    private static async Task<T> RunRequestAsync<T>(Func<Task<T>> request)
+    private static T RunRequest<T>(Func<T> request)
     {
         try
         {
-            return await request.Invoke().ConfigureAwait(false);
+            return request.Invoke();
         }
         catch (RequestFailedException e)
```

Comment:
New OpenAI client throws a `System.ClientModel`, `ClientResultException`, check for it.

```suggestion
        catch (ClientResultException cre)
```

Reply:
Yep, thanks, fixed.